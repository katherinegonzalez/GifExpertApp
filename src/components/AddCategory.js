import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (event) => {
        setInputValue(event.target.value);
        console.log('HandleInputChange llamado');
    }
    const handleSubmit = (event) => {
        event.preventDefault();  // Para prevenir el comportamiento por defecto del form, que en este caso serÃ¬a renderizar todo cuando se hace enter, pero no queremos que pase
        console.log('Submit hecho', inputValue);

        if(inputValue.trim().length > 2 ){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />

        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
