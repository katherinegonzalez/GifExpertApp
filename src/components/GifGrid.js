import React, { useState, useEffect } from 'react'

// useEffect -> me eprmite ejecutar cierto cÃ³digo de manera condicionada.

export const GifGrid = ({category}) => {

    const [images, setImages] = useState([]);

    useEffect(()=> {
        getGifs();
    }, []); // Se ejecuta cuando el componente se renderiza por primera vez

    const getGifs = async () => {
        
        const url = 'http://api.giphy.com/v1/gifs/search?q=Rick and Morty&limit=10&api_key=ehCxuH2BVdfrlfRwY0km2tRg2nwAw4LJ';
        const resp = await fetch(url);
        const { data } = await resp.json();

        
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
        setImages( gifs );
    }

    

    return (
        <div>
            <h3>{category}</h3>
            <ol>
            { 
                images.map(({id, title}) =>
                    <li key= {id}> {title } </li>
                )
            }
                
            </ol>
        </div>
    )
}
