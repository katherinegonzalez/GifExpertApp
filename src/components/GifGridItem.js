import React from 'react'

export const GifGridItem = ( {id, title, url }) => {
    console.log();
    return (
        <div className="card">
           <img src={ url } alt={ title } />
           <p>{title}</p>
        </div>
    )
}
