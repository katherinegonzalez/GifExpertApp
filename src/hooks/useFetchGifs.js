import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(()=> {
        getGifs( category )
            .then(imgs => setstate({
                data: imgs,
                loading: false
            })); // Los efectos no pueden ser async porque esperan algo s√≠ncrono, entonces uso .then
    }, [ category ]); // Se ejecuta cuando el componente se renderiza por primera vez
    // Si la category cambia entonces vuelve a ejecutar este efecto. para eso la ponemos enntre los []
    


    return state; // { data: [], loading: true }
}