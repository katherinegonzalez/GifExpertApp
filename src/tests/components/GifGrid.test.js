import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
import '@testing-library/jest-dom';

jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid />", () => {
  // let wrapper;
  const category = "One punch man";
  const gifs = [
    {
      id: "ABC",
      url: "https://localhost/cualquier/cosa.jpg",
      title: "Cualquier cosa",
    },
  ];

  /*beforeEach(() => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        wrapper = shallow(<GifGrid category={category}/>);
    });*/

  test("Debe mostrar el componente correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  });
});
