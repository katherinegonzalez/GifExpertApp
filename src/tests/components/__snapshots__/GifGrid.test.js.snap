// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch man
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cualquier cosa"
        url="https://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;
